nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/state/flow/flow.effects.ts

```diff
@@ -2056,4 +2058,277 @@ export class FlowEffects {
+                                map((response) => {
+                                    return FlowActions.startComponentSuccess({
+                                        response: {
+                                            type: request.type,
+                                            component: response
+                                        }
+                                    });
+                                })
+                            );
+                        }
+                        return of(
+                            FlowActions.flowApiError({
+                                error: `Starting ${request.type} requires both uri and revision properties`
+                            })
+                        );
+                    case ComponentType.ProcessGroup:
+                        return combineLatest([
+                            this.flowService.startProcessGroup(request),
+                            this.flowService.startRemoteProcessGroupsInProcessGroup(request)
+                        ]).pipe(
+                            map(([startPgResponse]) => {
+                                return FlowActions.startComponentSuccess({
+                                    response: {
+                                        type: request.type,
+                                        component: startPgResponse
+                                    }
+                                });
+                            })
+                        );
+                    default:
+                        return of(FlowActions.flowApiError({ error: `${request.type} does not support starting` }));
+                }
+            }),
+            catchError((error) => of(FlowActions.flowApiError({ error: error.error })))
+        )
+    );
+
+    /**
+     * If the component started was the current process group, reload the flow
+     */
+    startCurrentProcessGroupSuccess$ = createEffect(() =>
+        this.actions$.pipe(
+            ofType(FlowActions.startComponentSuccess),
+            map((action) => action.response),
+            withLatestFrom(this.store.select(selectCurrentProcessGroupId)),
+            filter(([response, currentPg]) => response.component.id === currentPg),
+            switchMap(() => of(FlowActions.reloadFlow()))
+        )
+    );
+
+    /**
+     * If a ProcessGroup was started, it should be reloaded as the response from the start operation doesn't contain all the displayed info
+     */
+    startProcessGroupSuccess$ = createEffect(() =>
+        this.actions$.pipe(
+            ofType(FlowActions.startComponentSuccess),
+            map((action) => action.response),
+            withLatestFrom(this.store.select(selectCurrentProcessGroupId)),
+            filter(([response, currentPg]) => response.component.id !== currentPg),
+            filter(([response]) => response.type === ComponentType.ProcessGroup),
```

Comment:
Nit-picky but could flip the ordering of the `filter` and only compare if the identifiers the `response.type` is `ProcessGroup`.